### ISSUE TITLE
EFAULT when queueing output buffers after reinitialisation

### ISSUE THREAD

**Issue by EwanRoycroft** [bbc/chromium#32]
This ticket is to track work to fix an issue seen in both #25 and #29, and when streaming the low-latency trial.

When reinitialising a stream, an `EFAULT` is returned from `VIDIOC_QBUF` requests for the output buffers. E.g.:
```
2021-08-02T15:03:27.353974Z VERBOSE1 chrome[1367:1499]: [v4l2_device.cc(1245)] QueueBuffer(): (CAPTURE_MPLANE) VIDIOC_QBUF failed: Bad address (14)
2021-08-02T15:03:27.354054Z ERROR chrome[1367:1499]: [v4l2_video_decoder_backend_stateful.cc(308)] Error while queueing output buffer!
2021-08-02T15:03:27.354092Z VERBOSE1 chrome[1367:1499]: [v4l2_video_decoder.cc(717)] SetState(): Error occurred, stopping queues.
```

**Comment by EwanRoycroft** [comment]
I've added the following line to `V4L2Queue::QueueBuffer()`:
```c++
VLOGF(1) << "v4l2_buffer memory address is: " << std::hex << (void *)v4l2_buffer;
```

It seems that the memory address for one of the output buffers is the same before and after the reinitialisation:
```
2021-11-24T18:16:49.999331Z VERBOSE1 chrome[6410:6567]: [v4l2_device.cc(1243)] QueueBuffer(): v4l2_buffer memory address is: 0x95d7680
2021-11-24T18:16:50.005910Z VERBOSE1 chrome[6410:6567]: [v4l2_video_decoder.cc(219)] Initialize(): Using a stateful API for profile: h264 high and fourcc: H264
2021-11-24T18:16:50.021988Z VERBOSE1 chrome[6410:6567]: [v4l2_device.cc(1243)] QueueBuffer(): v4l2_buffer memory address is: 0x95d7680
2021-11-24T18:16:50.022586Z VERBOSE1 chrome[6410:6567]: [v4l2_device.cc(1247)] QueueBuffer(): (CAPTURE_MPLANE) VIDIOC_QBUF failed: Bad address (14)
2021-11-24T18:16:50.022991Z ERROR chrome[6410:6567]: [v4l2_video_decoder_backend_stateful.cc(308)] Error while queueing output buffer!
2021-11-24T18:16:50.023517Z VERBOSE1 chrome[6410:6567]: [v4l2_video_decoder.cc(718)] SetState(): Error occurred, stopping queues.
```

This is interesting because the buffers were deallocated and the queue destroyed during the second initialisation:
https://github.com/bbc/chromium/blob/74f9770f9d323f59310eececc1db931d39c637ba/media/gpu/v4l2/v4l2_video_decoder.cc#L147-L169

Why do the new buffers have the same memory address?

**Comment by EwanRoycroft** [comment]
I wonder if the changes made in 74f9770f9d323f59310eececc1db931d39c637ba are causing this issue. The output (CAPTURE) queue isn't supposed to be streamed straight after initialisation, but we force it to anyway. This has no ill-effect on the first initialisation, but perhaps it's an issue for subsequent initialisations.

**Comment by EwanRoycroft** [comment]
Made the following changes:
```diff
diff --git a/media/gpu/v4l2/v4l2_video_decoder.cc b/media/gpu/v4l2/v4l2_video_decoder.cc
index 9adc719cf50b..8b11961d0782 100644
--- a/media/gpu/v4l2/v4l2_video_decoder.cc
+++ b/media/gpu/v4l2/v4l2_video_decoder.cc
@@ -130,6 +130,8 @@ void V4L2VideoDecoder::Initialize(const VideoDecoderConfig& config,
   DCHECK(state_ == State::kUninitialized || state_ == State::kDecoding);
   DVLOGF(3);
 
+  bool start_output_queue = state_ == State::kUninitialized;
+
   if (!can_use_decoder_) {
     VLOGF(1) << "Reached maximum number of decoder instances ("
              << kMaxNumOfInstances << ")";
@@ -247,7 +249,7 @@ void V4L2VideoDecoder::Initialize(const VideoDecoderConfig& config,
     return;
   }
 
-  if (output_queue_->AllocateBuffers(kNumOutputBuffers, V4L2_MEMORY_DMABUF) == 0) {
+  if (start_output_queue && output_queue_->AllocateBuffers(kNumOutputBuffers, V4L2_MEMORY_DMABUF) == 0) {
     VLOGF(1) << "Failed to allocate output buffer.";
     std::move(init_cb).Run(StatusCode::kV4l2FailedResourceAllocation);
     return;
@@ -257,7 +259,7 @@ void V4L2VideoDecoder::Initialize(const VideoDecoderConfig& config,
   // decoder, and doesn't hurt for the stateless one.
   // Start streaming output queue for RPi (bbc/chromium #11)
   // TODO(ewanr): Remove allocate/stream for output queue once #11 is fixed
-  if (!StartStreamV4L2Queue(true)) {
+  if (!StartStreamV4L2Queue(start_output_queue)) {
     VLOGF(1) << "Failed to start streaming.";
     std::move(init_cb).Run(StatusCode::kV4L2FailedToStartStreamQueue);
     return;
```

This only starts the output queue the first time the decoder is initialised. The result is no EFAULT errors, but the video stops decoding. This means that the old instance of the `V4L2Device` does not continue to run in the background, and it must be the new instance that causes the errors.

**Comment by EwanRoycroft** [comment]
Made the following additions to log buffer queues/dequeues:
```diff
diff --git a/media/gpu/v4l2/v4l2_device.cc b/media/gpu/v4l2/v4l2_device.cc
index 5b8cd2954f01..9a7c77178f09 100644
--- a/media/gpu/v4l2/v4l2_device.cc
+++ b/media/gpu/v4l2/v4l2_device.cc
@@ -1239,6 +1239,7 @@ base::Optional<V4L2WritableBufferRef> V4L2Queue::GetFreeBufferForFrame(
 bool V4L2Queue::QueueBuffer(struct v4l2_buffer* v4l2_buffer,
                             scoped_refptr<VideoFrame> video_frame) {
   DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
+  VQLOGF(3);
 
   int ret = device_->Ioctl(VIDIOC_QBUF, v4l2_buffer);
   if (ret) {
@@ -1257,6 +1258,7 @@ bool V4L2Queue::QueueBuffer(struct v4l2_buffer* v4l2_buffer,
 
 std::pair<bool, V4L2ReadableBufferRef> V4L2Queue::DequeueBuffer() {
   DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
+  VQLOGF(3);
 
   // No need to dequeue if no buffers queued.
   if (QueuedBuffersCount() == 0)
```

Which gives the following output:
```
2021-12-01T15:12:34.614497Z VERBOSE1 chrome[6008:6130]: [v4l2_video_decoder.cc(218)] Initialize(): Using a stateful API for profile: h264 high and fourcc: H264
2021-12-01T15:12:34.635462Z VERBOSE3 chrome[6008:6130]: [v4l2_device.cc(1242)] QueueBuffer(): (CAPTURE_MPLANE)
2021-12-01T15:12:34.635899Z VERBOSE1 chrome[6008:6130]: [v4l2_device.cc(1246)] QueueBuffer(): (CAPTURE_MPLANE) VIDIOC_QBUF failed: Bad address (14)
2021-12-01T15:12:34.636307Z ERROR chrome[6008:6130]: [v4l2_video_decoder_backend_stateful.cc(308)] Error while queueing output buffer!
2021-12-01T15:12:34.636607Z VERBOSE1 chrome[6008:6130]: [v4l2_video_decoder.cc(717)] SetState(): Error occurred, stopping queues.
2021-12-01T15:12:34.638072Z VERBOSE3 chrome[6008:6130]: [v4l2_device.cc(1242)] QueueBuffer(): (CAPTURE_MPLANE)
2021-12-01T15:12:34.638515Z VERBOSE1 chrome[6008:6130]: [v4l2_device.cc(1246)] QueueBuffer(): (CAPTURE_MPLANE) VIDIOC_QBUF failed: Bad address (14)
2021-12-01T15:12:34.638963Z ERROR chrome[6008:6130]: [v4l2_video_decoder_backend_stateful.cc(308)] Error while queueing output buffer!
2021-12-01T15:12:34.639116Z VERBOSE1 chrome[6008:6130]: [v4l2_video_decoder.cc(717)] SetState(): Error occurred, stopping queues.
2021-12-01T15:12:34.640024Z VERBOSE3 chrome[6008:6130]: [v4l2_device.cc(1242)] QueueBuffer(): (CAPTURE_MPLANE)
2021-12-01T15:12:34.640453Z VERBOSE1 chrome[6008:6130]: [v4l2_device.cc(1246)] QueueBuffer(): (CAPTURE_MPLANE) VIDIOC_QBUF failed: Bad address (14)
2021-12-01T15:12:34.640884Z ERROR chrome[6008:6130]: [v4l2_video_decoder_backend_stateful.cc(308)] Error while queueing output buffer!
2021-12-01T15:12:34.641197Z VERBOSE1 chrome[6008:6130]: [v4l2_video_decoder.cc(717)] SetState(): Error occurred, stopping queues.
2021-12-01T15:12:34.641829Z VERBOSE3 chrome[6008:6130]: [v4l2_device.cc(1242)] QueueBuffer(): (CAPTURE_MPLANE)
2021-12-01T15:12:34.642237Z VERBOSE1 chrome[6008:6130]: [v4l2_device.cc(1246)] QueueBuffer(): (CAPTURE_MPLANE) VIDIOC_QBUF failed: Bad address (14)
2021-12-01T15:12:34.642444Z ERROR chrome[6008:6130]: [v4l2_video_decoder_backend_stateful.cc(308)] Error while queueing output buffer!
2021-12-01T15:12:34.642818Z VERBOSE1 chrome[6008:6130]: [v4l2_video_decoder.cc(717)] SetState(): Error occurred, stopping queues.
2021-12-01T15:12:34.643320Z VERBOSE3 chrome[6008:6130]: [v4l2_device.cc(1242)] QueueBuffer(): (CAPTURE_MPLANE)
2021-12-01T15:12:34.643784Z VERBOSE1 chrome[6008:6130]: [v4l2_device.cc(1246)] QueueBuffer(): (CAPTURE_MPLANE) VIDIOC_QBUF failed: Bad address (14)
2021-12-01T15:12:34.643982Z ERROR chrome[6008:6130]: [v4l2_video_decoder_backend_stateful.cc(308)] Error while queueing output buffer!
2021-12-01T15:12:34.644399Z VERBOSE1 chrome[6008:6130]: [v4l2_video_decoder.cc(717)] SetState(): Error occurred, stopping queues.
2021-12-01T15:12:34.645154Z VERBOSE3 chrome[6008:6130]: [v4l2_device.cc(1242)] QueueBuffer(): (CAPTURE_MPLANE)
2021-12-01T15:12:34.645593Z VERBOSE1 chrome[6008:6130]: [v4l2_device.cc(1246)] QueueBuffer(): (CAPTURE_MPLANE) VIDIOC_QBUF failed: Bad address (14)
2021-12-01T15:12:34.646073Z ERROR chrome[6008:6130]: [v4l2_video_decoder_backend_stateful.cc(308)] Error while queueing output buffer!
2021-12-01T15:12:34.646552Z VERBOSE1 chrome[6008:6130]: [v4l2_video_decoder.cc(717)] SetState(): Error occurred, stopping queues.
2021-12-01T15:12:34.647305Z VERBOSE3 chrome[6008:6130]: [v4l2_device.cc(1242)] QueueBuffer(): (CAPTURE_MPLANE)
2021-12-01T15:12:34.647800Z VERBOSE1 chrome[6008:6130]: [v4l2_device.cc(1246)] QueueBuffer(): (CAPTURE_MPLANE) VIDIOC_QBUF failed: Bad address (14)
2021-12-01T15:12:34.648001Z ERROR chrome[6008:6130]: [v4l2_video_decoder_backend_stateful.cc(308)] Error while queueing output buffer!
2021-12-01T15:12:34.648612Z VERBOSE1 chrome[6008:6130]: [v4l2_video_decoder.cc(717)] SetState(): Error occurred, stopping queues.
2021-12-01T15:12:34.649155Z VERBOSE3 chrome[6008:6130]: [v4l2_device.cc(1242)] QueueBuffer(): (CAPTURE_MPLANE)
2021-12-01T15:12:34.649791Z VERBOSE1 chrome[6008:6130]: [v4l2_device.cc(1246)] QueueBuffer(): (CAPTURE_MPLANE) VIDIOC_QBUF failed: Bad address (14)
2021-12-01T15:12:34.650346Z ERROR chrome[6008:6130]: [v4l2_video_decoder_backend_stateful.cc(308)] Error while queueing output buffer!
2021-12-01T15:12:34.650502Z VERBOSE1 chrome[6008:6130]: [v4l2_video_decoder.cc(717)] SetState(): Error occurred, stopping queues.
2021-12-01T15:12:34.651284Z VERBOSE3 chrome[6008:6130]: [v4l2_device.cc(1242)] QueueBuffer(): (CAPTURE_MPLANE)
2021-12-01T15:12:34.651678Z VERBOSE1 chrome[6008:6130]: [v4l2_device.cc(1246)] QueueBuffer(): (CAPTURE_MPLANE) VIDIOC_QBUF failed: Bad address (14)
2021-12-01T15:12:34.652090Z ERROR chrome[6008:6130]: [v4l2_video_decoder_backend_stateful.cc(308)] Error while queueing output buffer!
2021-12-01T15:12:34.652239Z VERBOSE1 chrome[6008:6130]: [v4l2_video_decoder.cc(717)] SetState(): Error occurred, stopping queues.
2021-12-01T15:12:34.652961Z VERBOSE3 chrome[6008:6130]: [v4l2_device.cc(1242)] QueueBuffer(): (CAPTURE_MPLANE)
2021-12-01T15:12:34.653392Z VERBOSE1 chrome[6008:6130]: [v4l2_device.cc(1246)] QueueBuffer(): (CAPTURE_MPLANE) VIDIOC_QBUF failed: Bad address (14)
2021-12-01T15:12:34.654065Z ERROR chrome[6008:6130]: [v4l2_video_decoder_backend_stateful.cc(308)] Error while queueing output buffer!
2021-12-01T15:12:34.654218Z VERBOSE1 chrome[6008:6130]: [v4l2_video_decoder.cc(717)] SetState(): Error occurred, stopping queues.
2021-12-01T15:12:34.654890Z VERBOSE3 chrome[6008:6130]: [v4l2_device.cc(1242)] QueueBuffer(): (CAPTURE_MPLANE)
2021-12-01T15:12:34.655291Z VERBOSE1 chrome[6008:6130]: [v4l2_device.cc(1246)] QueueBuffer(): (CAPTURE_MPLANE) VIDIOC_QBUF failed: Bad address (14)
2021-12-01T15:12:34.655717Z ERROR chrome[6008:6130]: [v4l2_video_decoder_backend_stateful.cc(308)] Error while queueing output buffer!
2021-12-01T15:12:34.655849Z VERBOSE1 chrome[6008:6130]: [v4l2_video_decoder.cc(717)] SetState(): Error occurred, stopping queues.
2021-12-01T15:12:34.656731Z VERBOSE3 chrome[6008:6130]: [v4l2_device.cc(1242)] QueueBuffer(): (CAPTURE_MPLANE)
2021-12-01T15:12:34.657120Z VERBOSE1 chrome[6008:6130]: [v4l2_device.cc(1246)] QueueBuffer(): (CAPTURE_MPLANE) VIDIOC_QBUF failed: Bad address (14)
2021-12-01T15:12:34.657509Z ERROR chrome[6008:6130]: [v4l2_video_decoder_backend_stateful.cc(308)] Error while queueing output buffer!
2021-12-01T15:12:34.657838Z VERBOSE1 chrome[6008:6130]: [v4l2_video_decoder.cc(717)] SetState(): Error occurred, stopping queues.
2021-12-01T15:12:34.658064Z VERBOSE3 chrome[6008:6130]: [v4l2_device.cc(1242)] QueueBuffer(): (CAPTURE_MPLANE)
2021-12-01T15:12:34.658923Z VERBOSE1 chrome[6008:6130]: [v4l2_device.cc(1246)] QueueBuffer(): (CAPTURE_MPLANE) VIDIOC_QBUF failed: Bad address (14)
2021-12-01T15:12:34.659368Z ERROR chrome[6008:6130]: [v4l2_video_decoder_backend_stateful.cc(308)] Error while queueing output buffer!
2021-12-01T15:12:34.659518Z VERBOSE1 chrome[6008:6130]: [v4l2_video_decoder.cc(717)] SetState(): Error occurred, stopping queues.
2021-12-01T15:12:34.660251Z VERBOSE3 chrome[6008:6130]: [v4l2_device.cc(1242)] QueueBuffer(): (CAPTURE_MPLANE)
2021-12-01T15:12:34.660412Z VERBOSE1 chrome[6008:6130]: [v4l2_device.cc(1246)] QueueBuffer(): (CAPTURE_MPLANE) VIDIOC_QBUF failed: Bad address (14)
2021-12-01T15:12:34.661221Z ERROR chrome[6008:6130]: [v4l2_video_decoder_backend_stateful.cc(308)] Error while queueing output buffer!
2021-12-01T15:12:34.661529Z VERBOSE1 chrome[6008:6130]: [v4l2_video_decoder.cc(717)] SetState(): Error occurred, stopping queues.
2021-12-01T15:12:34.661939Z VERBOSE3 chrome[6008:6130]: [v4l2_device.cc(1242)] QueueBuffer(): (CAPTURE_MPLANE)
2021-12-01T15:12:34.662329Z VERBOSE1 chrome[6008:6130]: [v4l2_device.cc(1246)] QueueBuffer(): (CAPTURE_MPLANE) VIDIOC_QBUF failed: Bad address (14)
2021-12-01T15:12:34.662738Z ERROR chrome[6008:6130]: [v4l2_video_decoder_backend_stateful.cc(308)] Error while queueing output buffer!
2021-12-01T15:12:34.663051Z VERBOSE1 chrome[6008:6130]: [v4l2_video_decoder.cc(717)] SetState(): Error occurred, stopping queues.
2021-12-01T15:12:34.663493Z VERBOSE3 chrome[6008:6130]: [v4l2_device.cc(1242)] QueueBuffer(): (CAPTURE_MPLANE)
2021-12-01T15:12:34.663916Z VERBOSE1 chrome[6008:6130]: [v4l2_device.cc(1246)] QueueBuffer(): (CAPTURE_MPLANE) VIDIOC_QBUF failed: Bad address (14)
2021-12-01T15:12:34.664364Z ERROR chrome[6008:6130]: [v4l2_video_decoder_backend_stateful.cc(308)] Error while queueing output buffer!
2021-12-01T15:12:34.664728Z VERBOSE1 chrome[6008:6130]: [v4l2_video_decoder.cc(717)] SetState(): Error occurred, stopping queues.
2021-12-01T15:12:34.665164Z VERBOSE3 chrome[6008:6130]: [v4l2_device.cc(1242)] QueueBuffer(): (CAPTURE_MPLANE)
2021-12-01T15:12:34.665559Z VERBOSE1 chrome[6008:6130]: [v4l2_device.cc(1246)] QueueBuffer(): (CAPTURE_MPLANE) VIDIOC_QBUF failed: Bad address (14)
2021-12-01T15:12:34.666137Z ERROR chrome[6008:6130]: [v4l2_video_decoder_backend_stateful.cc(308)] Error while queueing output buffer!
2021-12-01T15:12:34.666493Z VERBOSE1 chrome[6008:6130]: [v4l2_video_decoder.cc(717)] SetState(): Error occurred, stopping queues.
2021-12-01T15:12:34.666955Z VERBOSE3 chrome[6008:6130]: [v4l2_device.cc(1242)] QueueBuffer(): (OUTPUT_MPLANE)
2021-12-01T15:12:34.668066Z VERBOSE3 chrome[6008:6130]: [v4l2_device.cc(1242)] QueueBuffer(): (OUTPUT_MPLANE)
2021-12-01T15:12:34.670360Z VERBOSE3 chrome[6008:6130]: [v4l2_device.cc(1242)] QueueBuffer(): (OUTPUT_MPLANE)
2021-12-01T15:12:34.671110Z VERBOSE3 chrome[6008:6130]: [v4l2_device.cc(1242)] QueueBuffer(): (CAPTURE_MPLANE)
2021-12-01T15:12:34.671509Z VERBOSE1 chrome[6008:6130]: [v4l2_device.cc(1246)] QueueBuffer(): (CAPTURE_MPLANE) VIDIOC_QBUF failed: Bad address (14)
2021-12-01T15:12:34.671914Z ERROR chrome[6008:6130]: [v4l2_video_decoder_backend_stateful.cc(308)] Error while queueing output buffer!
2021-12-01T15:12:34.672031Z VERBOSE1 chrome[6008:6130]: [v4l2_video_decoder.cc(717)] SetState(): Error occurred, stopping queues.
2021-12-01T15:12:34.672745Z VERBOSE3 chrome[6008:6130]: [v4l2_device.cc(1242)] QueueBuffer(): (CAPTURE_MPLANE)
2021-12-01T15:12:34.673121Z VERBOSE1 chrome[6008:6130]: [v4l2_device.cc(1246)] QueueBuffer(): (CAPTURE_MPLANE) VIDIOC_QBUF failed: Bad address (14)
2021-12-01T15:12:34.673506Z ERROR chrome[6008:6130]: [v4l2_video_decoder_backend_stateful.cc(308)] Error while queueing output buffer!
2021-12-01T15:12:34.673925Z VERBOSE1 chrome[6008:6130]: [v4l2_video_decoder.cc(717)] SetState(): Error occurred, stopping queues.
2021-12-01T15:12:34.674547Z VERBOSE3 chrome[6008:6130]: [v4l2_device.cc(1242)] QueueBuffer(): (CAPTURE_MPLANE)
2021-12-01T15:12:34.674967Z VERBOSE1 chrome[6008:6130]: [v4l2_device.cc(1246)] QueueBuffer(): (CAPTURE_MPLANE) VIDIOC_QBUF failed: Bad address (14)
2021-12-01T15:12:34.675410Z ERROR chrome[6008:6130]: [v4l2_video_decoder_backend_stateful.cc(308)] Error while queueing output buffer!
2021-12-01T15:12:34.675899Z VERBOSE1 chrome[6008:6130]: [v4l2_video_decoder.cc(717)] SetState(): Error occurred, stopping queues.
2021-12-01T15:12:34.676881Z VERBOSE3 chrome[6008:6130]: [v4l2_device.cc(1242)] QueueBuffer(): (CAPTURE_MPLANE)
2021-12-01T15:12:34.677277Z VERBOSE1 chrome[6008:6130]: [v4l2_device.cc(1246)] QueueBuffer(): (CAPTURE_MPLANE) VIDIOC_QBUF failed: Bad address (14)
2021-12-01T15:12:34.677652Z ERROR chrome[6008:6130]: [v4l2_video_decoder_backend_stateful.cc(308)] Error while queueing output buffer!
2021-12-01T15:12:34.677762Z VERBOSE1 chrome[6008:6130]: [v4l2_video_decoder.cc(717)] SetState(): Error occurred, stopping queues.
```

The first thing to note is that the OUTPUT queue also continues streaming, as some output buffers are queued further down.

Something else to note is that the CAPTURE queues begins streaming first after reinitialising This differs to the first initialisation. Here, the OUTPUT queue starts streaming first, and CAPTURE buffers don't begin queueing until after the resolution change:
```
2021-12-01T15:12:30.639629Z VERBOSE1 chrome[6008:6130]: [v4l2_video_decoder.cc(218)] Initialize(): Using a stateful API for profile: h264 baseline and fourcc: H264
2021-12-01T15:12:30.911620Z VERBOSE3 chrome[6008:6130]: [v4l2_device.cc(1242)] QueueBuffer(): (OUTPUT_MPLANE)
2021-12-01T15:12:30.912574Z VERBOSE3 chrome[6008:6130]: [v4l2_device.cc(1242)] QueueBuffer(): (OUTPUT_MPLANE)
2021-12-01T15:12:30.913872Z VERBOSE3 chrome[6008:6130]: [v4l2_device.cc(1242)] QueueBuffer(): (OUTPUT_MPLANE)
2021-12-01T15:12:30.915330Z VERBOSE3 chrome[6008:6130]: [v4l2_device.cc(1242)] QueueBuffer(): (OUTPUT_MPLANE)
2021-12-01T15:12:30.916838Z VERBOSE3 chrome[6008:6130]: [v4l2_device.cc(1261)] DequeueBuffer(): (OUTPUT_MPLANE)
2021-12-01T15:12:30.918563Z VERBOSE3 chrome[6008:6130]: [v4l2_device.cc(1261)] DequeueBuffer(): (OUTPUT_MPLANE)
2021-12-01T15:12:30.919580Z VERBOSE3 chrome[6008:6130]: [v4l2_device.cc(1261)] DequeueBuffer(): (OUTPUT_MPLANE)
2021-12-01T15:12:30.920725Z VERBOSE3 chrome[6008:6130]: [v4l2_device.cc(1261)] DequeueBuffer(): (OUTPUT_MPLANE)
2021-12-01T15:12:30.921982Z VERBOSE3 chrome[6008:6130]: [v4l2_device.cc(1242)] QueueBuffer(): (OUTPUT_MPLANE)
2021-12-01T15:12:30.923354Z VERBOSE3 chrome[6008:6130]: [v4l2_device.cc(1261)] DequeueBuffer(): (OUTPUT_MPLANE)
2021-12-01T15:12:30.924163Z VERBOSE3 chrome[6008:6130]: [v4l2_device.cc(1242)] QueueBuffer(): (OUTPUT_MPLANE)
2021-12-01T15:12:30.924943Z VERBOSE3 chrome[6008:6130]: [v4l2_device.cc(1261)] DequeueBuffer(): (OUTPUT_MPLANE)
2021-12-01T15:12:30.925623Z VERBOSE3 chrome[6008:6130]: [v4l2_device.cc(1242)] QueueBuffer(): (OUTPUT_MPLANE)
2021-12-01T15:12:30.926422Z VERBOSE3 chrome[6008:6130]: [v4l2_device.cc(1261)] DequeueBuffer(): (OUTPUT_MPLANE)
2021-12-01T15:12:30.928974Z VERBOSE1 chrome[6008:6130]: [v4l2_device.cc(995)] SetVisibleRect(): visible_rect=0,0 192x108
2021-12-01T15:12:30.930253Z VERBOSE1 chrome[6008:6130]: [v4l2_video_decoder.cc(388)] SetupOutputFormat(): buffer modifier: 0
2021-12-01T15:12:30.932685Z VERBOSE3 chrome[6008:6130]: [v4l2_device.cc(1242)] QueueBuffer(): (CAPTURE_MPLANE)
2021-12-01T15:12:30.935164Z VERBOSE3 chrome[6008:6130]: [v4l2_device.cc(1242)] QueueBuffer(): (CAPTURE_MPLANE)
2021-12-01T15:12:30.936409Z VERBOSE3 chrome[6008:6130]: [v4l2_device.cc(1242)] QueueBuffer(): (CAPTURE_MPLANE)
2021-12-01T15:12:30.937797Z VERBOSE3 chrome[6008:6130]: [v4l2_device.cc(1242)] QueueBuffer(): (CAPTURE_MPLANE)
2021-12-01T15:12:30.939551Z VERBOSE3 chrome[6008:6130]: [v4l2_device.cc(1242)] QueueBuffer(): (CAPTURE_MPLANE)
2021-12-01T15:12:30.940530Z VERBOSE3 chrome[6008:6130]: [v4l2_device.cc(1242)] QueueBuffer(): (CAPTURE_MPLANE)
```

**Comment by EwanRoycroft** [comment]
Can be closed with #11.
